### PR TITLE
HDDS-4873. Re-replication failure throws NPE

### PR DESCRIPTION
**Pull request by sky76093016**
## What changes were proposed in this pull request?

Increase the log that helps users to use.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-4873

## How was this patch tested?

No test.

### CODE REVIEW

**Comment by sky76093016** [comment]
@jojochuang Can you review this PR for me?Thank you.

**Review by adoroszlai — CHANGES_REQUESTED** [review]
Thanks @sky76093016 for working on this.

We should not simply log "please retry one more time", as container replication is performed by datanodes automatically.  It will be retried eventually when datanode receives the replication command from SCM again.

If download fails currently the code reaches the exception handler with NPE.  The only problem with this is that it's noisy, and we could handle the condition more gracefully.  I think if `tempTarFile == null`, it should simply set task status to failed, and skip the code in the `try` block.

**Comment by jojochuang** [comment]
what Attila said. Thanks for the review!

File: hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/replication/DownloadAndImportReplicator.java [hunk 1]
```diff
@@ -109,23 +109,26 @@ public void replicate(ReplicationTask task) {
...
-      task.setTransferredBytes(bytes);
-
-      importContainer(containerID, path);
-      LOG.info("Container {} is replicated successfully", containerID);
-      task.setStatus(Status.DONE);
-    } catch (Exception e) {
-      LOG.error("Container {} replication was unsuccessful.", containerID, e);
+    if (tempTarFile == null) {
       task.setStatus(Status.FAILED);
+    }else{
```
Comment (adoroszlai) [inline hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/replication/DownloadAndImportReplicator.java:114]:
```suggestion
    } else {
```

**Review by adoroszlai — APPROVED** [review]
Thanks @sky76093016 for updating the patch.

File: hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/replication/DownloadAndImportReplicator.java [hunk 2]
```diff
@@ -109,23 +109,26 @@ public void replicate(ReplicationTask task) {
...
-      LOG.info("Container {} is replicated successfully", containerID);
-      task.setStatus(Status.DONE);
-    } catch (Exception e) {
-      LOG.error("Container {} replication was unsuccessful.", containerID, e);
+    if (tempTarFile == null) {
       task.setStatus(Status.FAILED);
+    } else {
+      try {
+        //wait for the download. This thread pool is limiting the parallel
+        //downloads, so it's ok to block here and wait for the full download.
```
Comment (cku328) [inline hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/replication/DownloadAndImportReplicator.java:117]:
```suggestion
        // Wait for the download. This thread pool is limiting the parallel
        // downloads, so it's ok to block here and wait for the full download.
```
The comments in code should begin with a whitespace after the leading forward slashes.

**Review by cku328 — APPROVED** [review]
LGTM. Thanks @sky76093016 for working on this.

**Pull request merged**